sdk/appconfiguration/app-configuration/package.json

```diff
@@ -91,6 +91,7 @@
     "@azure/core-http": "^1.2.0",
     "@azure/core-paging": "^1.1.1",
     "@azure/core-tracing": "1.0.0-preview.11",
+    "@azure/core-util": "^1.0.0-beta.1",
```

Comment:
is appconfiguration in beta? if not, it should not take a dep on a beta.

Reply:
App-config is beta, but we may soon GA in the next couple of months.
Why is core-util in beta anyways? Can't we make that GA instead?

Reply:
Because we did not settle on the public surface for it yet.

appconfig is in beta but core-http and core-amqp are not IIRC and you added core-util as a dep there. My recommendation is to keep the local implementation of delay for now and have it moved to core-util in another PR.

Reply:
I really don't want to keep 3 copies of delay (core-http, core-amqp and app-config).

Reply:
@jeremymeng @chradek @richardpark-msft

Reply:
Yeah it is not ideal.

@sarangan12 had a similar issue with `delay` being used in autorest generated code but had to inline it there for now until core-util goes GA. 

@ramya-rao-a @xirzec do we have a plan for GAing core-util yet?

Reply:
I think we are still debating between package v.s. shared source code like storage-common or keyvault-common. For storage and keyvault, it could be that the common package doesn't provide any useful APIs outside of Storage/KeyVault usage so publishing packages for them doesn't make a lot of sense.  

In my opinion it's better to use package approach for @core-util, not the shared source approach since it provides useful API to other packages. @chradek mentioned that one reason he chose to have @azure/core-crypto is that using shared source would require each consuming package to add the browser mappings to shared browser files in package.json.  Plus there are some other additional maintenance to do (tsconfig for one)

Reply:
Let's chat more about `core-util` at next week's team meeting. /cc @ramya-rao-a

Reply:
> I really don't want to keep 3 copies of delay (core-http, core-amqp and app-config).

Why cannot App Config use the one in core-http?

Since this PR's main objective is around fixing an issue with the retry policy, can we stick to just that in this PR?

Reply:
@ramya-rao-a So, you're suggesting core-http and core-amqp would have delay? Can do that if weare fine with 2 copies.

(What's the point of core-util then?)

Reply:
https://github.com/Azure/azure-sdk-for-js/issues/15930

Reply:
> What's the point of core-util then

It is pending the on the conclusion on the discussion we had today on the subject of code sharing

Reply:
We now have two copies of (advanced) delay.. core-amqp and core-http.

Logged #15930 for core-util discussion.